src/node.js

```diff
@@ -283,16 +285,22 @@
     // Needs to be accessible from beyond this scope.
     process._tickCallback = _tickCallback;
     process._tickDomainCallback = _tickDomainCallback;
+    process._lateTick = lateTick;
 
     process._setupNextTick(tickInfo, _tickCallback, _runMicrotasks);
 
     _runMicrotasks = _runMicrotasks.runMicrotasks;
 
+    function drainLateTickQueue() {
+      while (lateTickQueue.length > 0) lateTickQueue.shift()();
+    }
+
     function tickDone() {
       if (tickInfo[kLength] !== 0) {
         if (tickInfo[kLength] <= tickInfo[kIndex]) {
           nextTickQueue = [];
           tickInfo[kLength] = 0;
+          drainLateTickQueue();
```

Comment:
I am confused, if I call a process.nextTIck at the fresh start of a turn with empty queue, that's top level and it will take some time for it to be called.